interfaces/sourcegen/sourcegen/_orchestrate.py

```diff
@@ -23,28 +23,51 @@ def format(self, record):
         return formatter.format(record)
 
 
-def generate_source(lang: str, out_dir: str=""):
+def generate_source(lang: str, out_dir: str = "", verbose = False) -> None:
     """Main entry point of sourcegen."""
     loghandler = logging.StreamHandler(sys.stdout)
     loghandler.setFormatter(CustomFormatter())
-    _logger.handlers.clear()
-    _logger.addHandler(loghandler)
-    _logger.setLevel(logging.DEBUG)
-    _logger.info(f"Generating {lang!r} source files...")
+    _LOGGER.handlers.clear()
+    _LOGGER.addHandler(loghandler)
+    _LOGGER.setLevel(logging.DEBUG if verbose else logging.INFO)
+
+    if not out_dir:
+        _LOGGER.critical("Aborting: sourcegen requires output folder information.")
+        exit(1)
 
     module = importlib.import_module(__package__ + "." + lang)
     root = Path(module.__file__).parent
     config = read_config(root / "config.yaml")
     templates = read_config(root / "templates.yaml")
-    ignore_files: List[str] = config.pop("ignore_files", [])
-    ignore_funcs: Dict[str, List[str]] = config.pop("ignore_funcs", {})
+    ignore_files: list[str] = config.pop("ignore_files", [])
+    ignore_funcs: dict[str, list[str]] = config.pop("ignore_funcs", {})
+
+    msg = f"Starting sourcegen for {lang!r} API"
+    _LOGGER.info(msg)
 
-    files = HeaderFileParser.from_headers(ignore_files, ignore_funcs)
+    if lang == "clib":
+        # prepare for generation of CLib headers in main processing step
+        files = HeaderFileParser.headers_from_yaml(ignore_files, ignore_funcs)
+    elif lang == "csharp":
+        # csharp parses existing (traditional) CLib header files
+        files = HeaderFileParser.headers_from_h(ignore_files, ignore_funcs)
+    else:
+        # generate CLib headers from YAML specifications as a preprocessing step
```

Comment:
I think this ought to be `elif lang == "yaml"` and a final `else` that indicates an error rather than silently defaulting to generating YAML output.

Reply:
The `YamlSourceGenerator` is only a stand-in for other APIs to be generated later. Instead of parsing CLib headers as currently done by the .NET API, the information can be obtained directly from the YAML configuration as a preprocessing step. This ensures that all information is consistent. (The `csharp` branch should be eliminated eventually.)

Reply:
Right, I understand why `csharp` is a special case until we change it to rely on the Doxygen/YAML input. What I'm less clear on is why running any target other than `clib` requires first generating the CLib source -- while you might need those headers to compile another interface, it's not a requirement to run `sourcegen`, is it? Why doesn't this all just simplify to:
```py
    if lang == "csharp":
        # csharp parses existing (traditional) CLib header files
        files = HeaderFileParser.headers_from_h(ignore_files, ignore_funcs)
    else:
        files = HeaderFileParser.headers_from_yaml(ignore_files, ignore_funcs)
```

Reply:
@speth ... thinking about this a little more, the best approach would be to be more explicit about splitting the preprocessing step from the CLib generator. That way, we could use the `if` - `else` block you mentioned above and clarify some of the logic. Given the size of this PR, I'd prefer to do it investigate in a follow-up PR.